src/components/Settings/RadarrModal/index.tsx

```diff
@@ -130,7 +130,10 @@ const RadarrModal = ({ onClose, radarr, onSave }: RadarrModalProps) => {
       intl.formatMessage(messages.validationMinimumAvailabilityRequired)
     ),
     externalUrl: Yup.string()
-      .url(intl.formatMessage(messages.validationApplicationUrl))
+      .matches(
+        /^https?:\/\/(www\.)?[-a-zA-Z0-9@:%._+~#=]{1,256}(\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_+.~#?&//=]*))?$/i,
```

Comment:
What do you think of

```suggestion
        /^https?:\/\/(([-a-zA-Z0-9._+%]+(:[-a-zA-Z0-9._+%]+)?@)?([-a-zA-Z0-9._]+)(\.[a-zA-Z]{2,6})?)(:\d{1,5})?(\/[-a-zA-Z0-9()@:%_+.~#?&/=]*)?$/i,
```

(My best effort to not allow special characters in domain name as this could be syntaxically invalid [correct me if im wrong] but still allow basic auth to be included in the url)

Tested this using:
https://regexr.com/

Feel free to resolve this conversation if im wrong about it or made a mistake in my regex

Reply:
I took the regex from here: https://stackoverflow.com/a/3809435, and just edited it so it's not mandatory to have a TLD.
Here is the thing i tested:

```
http://domain
http://domain.com
http://domain.com:9999
http://domain.com/base
http://domain.com:9999/base
http://x.x.x.x:9999
```

The two regex match these URLs. Do you have any example of something your regex is trying to solve than mine doesn't?

Reply:
http://user:pass@example.com

Also not to allow
http://foo@bar.com
Or
http://foo#bar.com

Reply:
http://foo@bar.com/ is allowed by your regex